cmd/srcd/cmd/sql.go

```diff
@@ -121,24 +126,39 @@ func repl() error {
 			return nil
 		case "":
 		default:
-			if err := runQuery(statement); err != nil {
+			if err := runQuery(client, statement); err != nil {
 				fmt.Println(err)
 			}
 		}
 	}
 }
 
-func runQuery(query string) error {
-	c, err := daemon.Client()
-	if err != nil {
-		return fmt.Errorf("could not get daemon client: %v", err)
+func ensureConnReady(client api.EngineClient) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	sleep := 1
+	for {
+		_, err := client.SQL(ctx, &api.SQLRequest{Query: "SELECT 1"})
+		if err == nil {
+			logrus.Info("SQL connection ready")
+			break
+		}
+
+		logrus.Warningf("SQL connection not ready, trying in %d seconds", sleep)
+		time.Sleep(time.Duration(sleep) * time.Second)
+		sleep *= 2
```

Comment:
Is this function always returning no error?
Does it make sense to wait forever?
Does it make sense to increase the sleep by the double without no limit?

Reply:
> Is this function always returning no error?
> Does it make sense to wait forever?

If you use the CLI, I guess it's ok to wait forever since you can `ctrl+c` at any moment.
But if you call `srcd sql "query"` from a script, this can be bad. I would add a timeout, we should not wait forever.

About the `sleep *= 2`; I think this can grow too quick, and we could be waiting for the connection much more time that really needed.

I would go for a sensible fixed interval plus a global timeout.